healthcheck/probe_test.go

```
@@ -0,0 +1,323 @@
+	// Boundary values must be accepted.
+	for _, d := range []time.Duration{time.Nanosecond, MaxTimeout} {
+		srv := rpcServer(t, http.StatusOK, validHealthBody)
+		if err := Live(context.Background(), srv.URL, d); err != nil && errors.Is(err, errTimeoutRange) {
+			t.Fatalf("timeout %v must be accepted, got %v", d, err)
+		}
+	}
+}
+
+func TestUnreachable(t *testing.T) {
+	// Bind and immediately release a port so nothing listens on it.
+	ln, err := net.Listen("tcp", "127.0.0.1:0")
```

golangci-lint (2.12.2)

[error] net.Listen must not be called. use (*net.ListenConfig).Listen

(noctx)